File: src/Amalgam.Core/Configuration/RepositoryConfig.cs
namespace Amalgam.Core.Configuration;

public class RepositoryConfig
{
    public string Name { get; set; } = string.Empty;
    public RepositoryType Type { get; set; }
    public string Path { get; set; } = string.Empty;
    public bool Enabled { get; set; } = true;
    public string? RoutePrefix { get; set; }
    public string? PackageName { get; set; }
    public MergeConfig? Merge { get; set; }
}

File: src/Amalgam.Web/projects/api/src/lib/models/repository-config.ts
import { MergeConfig } from './merge-config';
import { RepositoryType } from './repository-type';

export interface RepositoryConfig {
  name: string;
  type: RepositoryType;
  path: string;
  enabled: boolean;
  routePrefix?: string;
  packageName?: string;
  merge?: MergeConfig;
}
